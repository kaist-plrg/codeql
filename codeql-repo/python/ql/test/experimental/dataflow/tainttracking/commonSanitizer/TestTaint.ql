import experimental.dataflow.tainttracking.TestTaintLib
import semmle.python.dataflow.new.BarrierGuards

class CustomSanitizerOverrides extends TestTaintTrackingConfiguration {
  override predicate isSanitizerGuard(DataFlow::BarrierGuard guard) {
    guard instanceof StringConstCompare
  }
}
