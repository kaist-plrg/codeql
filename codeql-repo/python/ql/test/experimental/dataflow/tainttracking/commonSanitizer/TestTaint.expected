| test_string_const_compare.py:16 | ok   | test_eq | ts |
| test_string_const_compare.py:18 | ok   | test_eq | ts |
| test_string_const_compare.py:20 | ok   | test_eq | ts |
| test_string_const_compare.py:27 | ok   | test_eq_unsafe | ts |
| test_string_const_compare.py:29 | ok   | test_eq_unsafe | ts |
| test_string_const_compare.py:35 | fail | test_eq_with_or | ts |
| test_string_const_compare.py:37 | ok   | test_eq_with_or | ts |
| test_string_const_compare.py:43 | ok   | test_non_eq1 | ts |
| test_string_const_compare.py:45 | ok   | test_non_eq1 | ts |
| test_string_const_compare.py:51 | ok   | test_non_eq2 | ts |
| test_string_const_compare.py:53 | fail | test_non_eq2 | ts |
| test_string_const_compare.py:59 | ok   | test_in_list | ts |
| test_string_const_compare.py:61 | ok   | test_in_list | ts |
| test_string_const_compare.py:67 | ok   | test_in_tuple | ts |
| test_string_const_compare.py:69 | ok   | test_in_tuple | ts |
| test_string_const_compare.py:75 | ok   | test_in_set | ts |
| test_string_const_compare.py:77 | ok   | test_in_set | ts |
| test_string_const_compare.py:83 | ok   | test_in_unsafe1 | ts |
| test_string_const_compare.py:85 | ok   | test_in_unsafe1 | ts |
| test_string_const_compare.py:91 | ok   | test_in_unsafe2 | ts |
| test_string_const_compare.py:93 | ok   | test_in_unsafe2 | ts |
| test_string_const_compare.py:99 | ok   | test_not_in1 | ts |
| test_string_const_compare.py:101 | ok   | test_not_in1 | ts |
| test_string_const_compare.py:107 | ok   | test_not_in2 | ts |
| test_string_const_compare.py:109 | fail | test_not_in2 | ts |
| test_string_const_compare.py:119 | fail | test_eq_thorugh_func | ts |
| test_string_const_compare.py:121 | ok   | test_eq_thorugh_func | ts |
